File: NATSInternal.Frontend/NATSInternal.React/src/models/authentication/signInModel.ts
declare global {
  type SignInModel = {
    userName: string;
    password: string;
    toRequestDto(): VerifyUserNameAndPasswordRequestDto;
  };
}

export function createSignInModel(): SignInModel {
  return {
    userName: "",
    password: "",
    toRequestDto(): VerifyUserNameAndPasswordRequestDto {
      return {
        userName: this.userName,
        password: this.password
      };
    }
  };
}

File: NATSInternal.Api/Models/Authentication/SignInModel.cs
using NATSInternal.Application.UseCases.Authentication;

namespace NATSInternal.Api.Models;

public class SignInModel
{
    #region Properties
    public required string UserName { get; set; }
    public required string Password { get; set; }
    #endregion

    #region Methods
    public VerifyUserNameAndPasswordRequestDto ToRequestDto()
    {
        return new()
        {
            UserName = UserName,
            Password = Password
        };
    }
    #endregion
}